packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts

```diff
@@ -422,51 +416,387 @@ describe('Appointment Popup Form', () => {
+          currentView: 'month',
+          currentDate: new Date(2017, 4, 25),
+          firstDayOfWeek: 1,
+          startDayHour: 9,
+        });
+
+        POM.openPopupByDblClick('Weekly Meeting');
+
+        POM.popup.getEditSeriesButton().click();
+
+        // @ts-expect-error
+        const repeatEditorRecurrent = $(POM.popup.repeatEditor).dxSelectBox('instance');
+        const buttons = repeatEditorRecurrent.option('buttons');
+
+        expect(repeatEditorRecurrent.option('value')).toBe('weekly');
+
+        const settingsButton = buttons?.find((btn) => btn.name === 'settings');
+        if (settingsButton?.options?.onClick) {
+          settingsButton.options.onClick();
+        }
+
+        // @ts-expect-error
+        const freqEditor = $(POM.popup.freqEditor).dxSelectBox('instance');
+        expect(freqEditor.option('value')).toBe('weekly');
+
+        const dayButtons = $(POM.popup.element)
+          .find('.dx-scheduler-recurrence-byday-buttons .dx-button');
+        const selectedButtons = dayButtons.filter('.dx-button-mode-contained');
+        expect(selectedButtons.length).toBe(3);
+
+        const repeatEndEditor = POM.popup.getSwitchByName('repeatEnd');
+        expect(repeatEndEditor.value).toBe('count');
+
+        const countEditor = POM.popup.getSwitchByName('count');
+        expect(countEditor.value).toBe('10');
+      });
+
+      it('Should save changes when saving from recurrence form', async () => {
+        const data = [
+          {
+            text: 'Meeting',
+            startDate: new Date(2017, 4, 22, 9, 30),
+            endDate: new Date(2017, 4, 22, 11, 30),
+          },
+        ];
+
+        const { POM, scheduler } = await createScheduler({
+          dataSource: data,
+          currentDate: new Date(2017, 4, 22),
+        });
+
+        POM.openPopupByDblClick('Meeting');
+
+        // @ts-expect-error
+        const repeatEditor = $(POM.popup.repeatEditor).dxSelectBox('instance');
+        repeatEditor.option('value', 'daily');
+
+        const buttons = repeatEditor.option('buttons');
+        const settingsButton = buttons?.find((btn) => btn.name === 'settings');
+        if (settingsButton?.options?.onClick) {
+          settingsButton.options.onClick();
+        }
+
+        // @ts-expect-error
+        const freqEditor = $(POM.popup.freqEditor).dxSelectBox('instance');
+        freqEditor.option('value', 'weekly');
+
+        POM.popup.getBackButton().click();
+
+        scheduler.hideAppointmentPopup(true);
+
+        POM.openPopupByDblClick('Meeting');
+
+        POM.popup.getEditSeriesButton().click();
+
+        // @ts-expect-error
+        const freqEditorAfter = $(POM.popup.freqEditor).dxSelectBox('instance');
+        expect(freqEditorAfter.option('value')).toBe('weekly');
+
+        // @ts-expect-error
+        const repeatEditorAfter = $(POM.popup.repeatEditor).dxSelectBox('instance');
+        expect(repeatEditorAfter.option('value')).toBe('weekly');
```

Comment:
It's better to check that actual appointment's data changed by checking it from data source rather than checking it in form, because form could have saved previous appointments state.